### PR TITLE
boost: update to 1.68.0.

### PR DESCRIPTION
**Pull request by Johnnynator**
|Package|x86_64|x86_64-musl|i686|aarch64|aarch64-musl|armv7l|armv7l-musl|armv6l|armv6l-musl|/|
|-------|------|-----------|----|-------|------------|------|-----------|------|-----------|-----|
|boost| c | c | c | c | c | | c | c | | |
|volk| c | c | | | | | | | | |
|openimageio| c | c | | | | | | | | |
|gnuradio| c | c | | | | | | | | |
|leatherman| c | c | | | | | | | | |
|Lucene++| c | c | | | | | | | | |
|synfig| c | | | | | | | | | |
|monero| c | c | | | | | | | | |
|luxrays| c | | | | | | | | | |
|libixion| c  | c | | | | | | | | |
|gnuradio-osmosdr| c | c | | | | | | | | |
|libtorrent-rasterbar| c | c | | | | | | | | |
|cegui07| c | c | | | | | | | | |
|widelands| c | c | | | | | | | | |
|wesnoth| c | c | | | | | | | | |
|usbtop| c | c | | | | | | | | |
|tomahawk| c | c | | | | | | | | |
|tmfs| c | c | | | | | | | | |
|swiften| c | c | | | | | | | | |
|supercollider| c | | | | | | | | | |
|strigi| c | | | | | | | | | |
|source-highlight| c | c | | | | | | | | |
|snapper| c | c | | | | | | | | |
|schroot| c | c | | | | | | | | |
|scantailor| c | c | | | | | | | | |
|rstudio| c | | | | | | | | | |
|pulseview| c | c | | | | | | | | |
|povray| c | / | | | | | | | | |
|poedit| c | | | | | | | | | |
|pingus| c | | | | | | | | | |
|pamixer| c | | | | | | | | | |
|openmw| c | | | | | | | | | |
|omxplayer|only arm | | | | | | c | | | |
|ncmpcpp| c | | | | | | | | | |
|namecoin| c | | | | | | | | | |
|mupen64plus| c | | | | | | | | | |
|mongodb| c | | | | | | | | | |
|monero-gui| c | | | | | | | | | |
|mkvtoolnix| c | | | | | | | | | |
|logstalgia| c | | | | | | | | | |
|litecoin| c | | | | | | | | | |
|librime| c | | | | | | | | | |
|liborcus| c | | | | | | | | | |
|libogre| c | | | | | | | | | |
|libcmis| c | c | | | | | | | | |
|ledger| c | | | | | | | | | |
|krita| c | | | | | | | | | |
|kig| c | | | | | | | | | |
|kea| c | | | | | | | | | |
|innoextract| c | | | | | | | | | |
|imagescan| c | | | | | | | | | |
|icinga2| c | | | | | | | | | |
|i2pd| c | | | | | | | | | |
|hugin| c | | | | | | | | | |
|guitarix2| c | | | | | | | | | |
|grive2| c | | | | | | | | | |
|gqrx| c | | | | | | | | | |
|gpick| c | | | | | | | | | |
|gource| c | | | | | | | | | |
|gnucash| c | | | | | | | | | |
|freeorion| c | | | | | | | | | |
|freecad| c | | | | | | | | | |
|fifengine| c | | | | | | | | | |
|facter| c | | | | | | | | |
|eiskaltdcpp-qt| c | | | | | | | | | |
|cgal| c | | | | | | | | | |
|cegui| c | | | | | | | | | |
|cclive| c | | | | | | | | | |
|btfs| c | | | | | | | | | |
|blender| c | | | | | | | | | |
|bitcoin| c | | | | | | | | | |
|astroid| c | | | | | | | | | |
|ardour| c | | | | | | | | | |
|adriconf| c | | | | | | | | | |
|TSC| c | | | | | | | | | |
|SMC| c | | | | | | | | | |
|OpenXcom| c | | | | | | | | | |
|LGOGDownloader| c | | | | | | | | | |
|EmulationStation-rpi| / | | | | | c | c | | | |
|EmulationStation| c | | | | | | | | | |
|Aegisub| c | | | | | | | | | |
|0ad| c | | | | | | | | | |
|libreoffice| c | | | | | | | | | |
|qbittorrent| c | | | | | | | | | |

### CODE REVIEW

**Comment by maxice8** [comment]
openlierox had it's boost dep removed

**Comment by maxice8** [comment]
yaml-cpp 0.62 dropped the boost dependency

https://github.com/jbeder/yaml-cpp/releases

**Comment by Johnnynator** [comment]
@pullmoll @maxice8 Can anyone of you two help me with facter. I get some linking errors which I'm not sure how to solve them. https://gist.github.com/Johnnynator/4419fda968bc25c8ac96c72478b2670a

**Comment by maxice8** [comment]
On Thu, Oct 04, 2018 at 02:23:06PM -0700, Johannes wrote:
> Bumping leatherman before building facter solves most of the errors.
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/3230#issuecomment-427174193

problably the stdc++ thing ?

**Comment by jnbr** [comment]
no, just the boost update (bump is missing in the commits)

**Comment by Johnnynator** [comment]
@jnbr OK thanks for checking it out. I could have sworn what I already revbumped leatherman but it probably got lost at some point.

**Comment by Johnnynator** [comment]
So for now only boost-python3 and mongodb are missing.
Is it ok to switch mongodb to it's own boost version, I don't have any interest to patch this mongodb version.

**Comment by Johnnynator** [comment]
It is fine for review now. Every package compiled fine for x86_64, other arches only tested compile time randomly and haven't encountered any issues so far.

**Comment by jnbr** [comment]
tomahawk fails on aarch64-musl

**Comment by Johnnynator** [comment]
> tomahawk fails to link on aarch64-musl
> note: it fails without the boost update as well

Yep. It is missing `libsecret-devel`. Seems to work with it. I updated the template.

**Comment by jnbr** [comment]
supercollider fails as well, uses invalid compiler options for aarch64

**Comment by Johnnynator** [comment]
> supercollider fails as well, uses invalid compiler options for aarch64

I see. I'm wondering why the architecture check in the cmake file now fails and worked before. I just added `-DSSE=OFF -DSSE2=OFF` for all non x86 builds for now.

**Comment by jnbr** [comment]
Now everything compiles fine on aarch64-musl and armv7l.

File: srcpkgs/boost/template [hunk 1]
```diff
@@ -37,9 +32,17 @@ do_build() {
 	LIBS="$LDFLAGS" ${wrksrc}/bjam -f build.jam --toolset=cc --toolset-root= -d+2 clean
 	LIBS="$LDFLAGS" ${wrksrc}/bjam -f build.jam --toolset=cc --toolset-root= -d+2
 	)
+	
+	cat > user-config.jam <<-__EOF
+	using gcc : ${_toolset#gcc-} : ${CXX} : <cxxflags>"${CXXFLAGS}" <linkflags>"${LDFLAGS}" ;
+	using python : 2.7 : ${XBPS_CROSS_BASE}/usr/bin/python : ${XBPS_CROSS_BASE}/usr/include/python2.7 : ${XBPS_CROSS_BASE}/usr/lib/python2.7 ;
```
Comment (asergi) [inline srcpkgs/boost/template]: `'s|${XBPS_CROSS_BASE}/usr/bin/python|${XBPS_CROSS_BASE}/usr/bin/python2|'`

File: srcpkgs/boost/template [hunk 2]
```diff
@@ -37,9 +32,17 @@ do_build() {
...
+	using python : 3.6 : ${XBPS_CROSS_BASE}/usr/bin/python3 : ${XBPS_CROSS_BASE}/usr/include/python3.6m : ${XBPS_CROSS_BASE}/usr/lib/python3.6 ;
```
Comment (asergi) [inline srcpkgs/boost/template]: Please do not hardcode the Python 3.x version.
Better to open a separate PR for `boost-python3`.

Comment (Johnnynator) [reply srcpkgs/boost/template]: Ok I split it up into it's own PR.

**Comment by Gottox** [comment]
Ready to merge?

**Comment by Johnnynator** [comment]
> The builder is busy anyway :)

OK. I did rebase it nevertheless. Makes it easier if it needs to be rebased later.

**Pull request merged**